app/src/main/java/com/douglasbruce/tasky/features/agenda/AgendaViewModel.kt

```diff
@@ -100,19 +101,28 @@ class AgendaViewModel @Inject constructor(
 
             is AgendaEvent.OnDeleteEventClick -> {
                 viewModelScope.launch {
-                    eventRepository.deleteEventById(event.eventId)
+                    val result = eventRepository.deleteEventById(event.eventId)
+                    if (result is AuthResult.Unauthorized) {
+                        state = state.copy(logout = true)
+                    }
                 }
             }
 
             is AgendaEvent.OnDeleteTaskClick -> {
                 viewModelScope.launch {
-                    taskRepository.deleteTaskById(event.taskId)
+                    val result = taskRepository.deleteTaskById(event.taskId)
+                    if (result is AuthResult.Unauthorized) {
+                        state = state.copy(logout = true)
+                    }
```

Comment:
Technically, you could also have a global logout StateFlow which you update from within an interceptor if a request returns 401. That can then be observed by your app's root UI and directly navigate the user back to the login screen. This way, you only need to handle the logic at a single place, so you have less risk forgetting this somewhere